MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.h

```diff
@@ -77,6 +77,61 @@ typedef enum : uint16_t {
 	kMVKMTLFmtCapsMultiPlanar = kMVKMTLFmtCapsChromaSubsampling,
 } MVKMTLFmtCaps;
 
+inline MVKMTLFmtCaps operator|(MVKMTLFmtCaps leftCaps, MVKMTLFmtCaps rightCaps) {
+	return static_cast<MVKMTLFmtCaps>(static_cast<uint32_t>(leftCaps) | rightCaps);
+}
+
+inline MVKMTLFmtCaps& operator|=(MVKMTLFmtCaps& leftCaps, MVKMTLFmtCaps rightCaps) {
+	return (leftCaps = leftCaps | rightCaps);
+}
+
+
+#pragma mark -
+#pragma mark Metal view classes
+
+enum class MVKMTLViewClass : uint8_t {
+	None,
+	Color8,
+	Color16,
+	Color32,
+	Color64,
+	Color128,
+	PVRTC_RGB_2BPP,
+	PVRTC_RGB_4BPP,
+	PVRTC_RGBA_2BPP,
+	PVRTC_RGBA_4BPP,
+	EAC_R11,
+	EAC_RG11,
+	EAC_RGBA8,
+	ETC2_RGB8,
+	ETC2_RGB8A1,
+	ASTC_4x4,
+	ASTC_5x4,
+	ASTC_5x5,
+	ASTC_6x5,
+	ASTC_6x6,
+	ASTC_8x5,
+	ASTC_8x6,
+	ASTC_8x8,
+	ASTC_10x5,
+	ASTC_10x6,
+	ASTC_10x8,
+	ASTC_10x10,
+	ASTC_12x10,
+	ASTC_12x12,
+	BC1_RGBA,
+	BC2_RGBA,
+	BC3_RGBA,
+	BC4_R,
+	BC5_RG,
+	BC6H_RGB,
+	BC7_RGBA,
+	Depth24_Stencil8,
+	Depth32_Stencil8,
+	BGRA10_XR,
+	BGR10_XR
+};
+
```

Comment:
Looks like `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT` was introduced in `VK_KHR_maintenance2`, which also introduced uncompressed image views of compressed images.

Does this classification scheme allow, say, a 64-bit compressed block to adopt the capabilities of a 64-bit color format? Or what about between two compressed formats of the same block size? It seems more divided than that.

Or am I missing something in my understanding?

Reply:
> Does this classification scheme allow, say, a 64-bit compressed block to adopt the capabilities of a 64-bit color format? Or what about between two compressed formats of the same block size? It seems more divided than that.

Deliberately not, since Metal still doesn't allow reinterpreting a compressed texture with an uncompressed format.

Reply:
Meh. You're right of course. I brain-farted between Metal and Vulkan capabilities.